### PR TITLE
Add End2End PipelineRun test

### PR DESCRIPTION
**Pull request by CarolynMabbott**
# Changes

References issue #161 

Add end2end test by creating a Pipelinerun. Creates a svc, which is checked to confirm it exists and is running as expected via curling the pod. 

Includes error checking and meaningful messages for test failures 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by CarolynMabbott** [comment]
I signed it!

File: test/e2e-tests.sh [hunk 1]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
 install_pipeline_crd
 install_dashboard_backend
 
-failed=0
+# failed=0
```
Comment (a-roberts) [inline test/e2e-tests.sh]: This is commented out but later on we use it (( failed )), so does this definitely work?

File: test/e2e-tests.sh [hunk 2]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
 
 # Run the integration tests
-header "Running e2e tests"
-# TODO: run your test here !
+ header "Running e2e tests"
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Whitespace

File: test/e2e-tests.sh [hunk 3]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+IMAGE_SOURCE_NAME="docker-image"
+GIT_RESOURCE_NAME="git-source"
+GIT_COMMIT="master"
+REPO_NAME="go-hello-world"
+REPO_URL="https://github.com/ncskier/go-hello-world" 
+EXPECTED_RETURN_VALUE="Hello World! "
+KSVC_NAME="go-hello-world"
+REGISTRY="gcr.io/${E2E_PROJECT_ID}/${E2E_BASE_NAME}-e2e-img"
+
+   post_data='{
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Whitespace

File: test/e2e-tests.sh [hunk 4]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+do
+   respF=$(curl -k  http://127.0.0.1:9097)
+   if [ "$respF" != "" ]; then
+        dashboardExists=true
+        break
+    else    
+        sleep 5  
+    fi
+  
+done
```
Comment (a-roberts) [inline test/e2e-tests.sh]: I'd move this up a line so it's clearly part of the loop

File: test/e2e-tests.sh [hunk 5]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+if [ "$dashboardExists" = "false" ]; then
+        fail_test "Test Failure, Not able to curl the dashboard"
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Whitespace and casing

File: test/e2e-tests.sh [hunk 6]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+fi 
+
+curlNport="http://127.0.0.1:9097/v1/namespaces/${APP_NS}/pipelineruns/"
```
Comment (a-roberts) [inline test/e2e-tests.sh]: curlNodePort would be easier to interpret

File: test/e2e-tests.sh [hunk 7]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+do
+   wait=$(kubectl wait --for=condition=available deployments/go-hello-world --timeout=30s)
+   if [ "$wait" = "deployment.extensions/go-hello-world condition met" ]; then
+        deploymentExist=true
+        break
+    else    
+        sleep 5  
+    fi 
+done
+if [ "$deploymentExist" = "false" ]; then
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Whitespace here after the done

File: test/e2e-tests.sh [hunk 8]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+fi 
+
+curlNport="http://127.0.0.1:9097/v1/namespaces/${APP_NS}/pipelineruns/"
+curl -X POST --header Content-Type:application/json -d "$post_data" $curlNport 
+
+deploymentExist=false
+for i in {1..30}
+do
+   wait=$(kubectl wait --for=condition=available deployments/go-hello-world --timeout=30s)
+   if [ "$wait" = "deployment.extensions/go-hello-world condition met" ]; then
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Nice 😄

File: test/e2e-tests.sh [hunk 9]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+        deploymentExist=true
+        break
+    else    
+        sleep 5  
+    fi 
+done
+if [ "$deploymentExist" = "false" ]; then
+        fail_test "Test Failure, go-hello-world deployment is not running"
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Whitespace

File: test/e2e-tests.sh [hunk 10]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+   if [ "$resp" != "" ]; then
+        podCurled=true
+        if [ "$EXPECTED_RETURN_VALUE" = "$resp" ]; then
+     echo "Pipeline Run successfully executed"
+     break
+    else
+     fail_test "Pipeline Run error returned not expected message: $resp"
+    fi 
+    else    
+        sleep 5
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Whitespace

File: test/e2e-tests.sh [hunk 11]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+if [ "$deploymentExist" = "false" ]; then
+        fail_test "Test Failure, go-hello-world deployment is not running"
+fi 
+
+kubectl port-forward $(kubectl get pod -l app=go-hello-world -o name) 8080 &
+
+podCurled=false
+for i in {1..20}
+do
+    resp=$(curl -k  http://127.0.0.1:8080)
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Whitespace

File: test/e2e-tests.sh [hunk 12]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+   if [ "$resp" != "" ]; then
+        podCurled=true
+        if [ "$EXPECTED_RETURN_VALUE" = "$resp" ]; then
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Whitespace

File: test/e2e-tests.sh [hunk 13]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+     echo "Pipeline Run successfully executed"
```
Comment (a-roberts) [inline test/e2e-tests.sh]: PipelineRun

File: test/e2e-tests.sh [hunk 14]
```diff
@@ -28,11 +28,98 @@ header "Setting up environment"
...
+     break
+    else
+     fail_test "Pipeline Run error returned not expected message: $resp"
```
Comment (a-roberts) [inline test/e2e-tests.sh]: PipelineRun, and grammar (doesn't really make sense when you read this)

**Comment by a-roberts** [comment]
Tests pass and output looks tidy enough, thanks @CarolynMabbott!

/approve
/lgtm

**Pull request merged**